public/app/features/apiserver/server.ts

```diff
@@ -48,7 +53,13 @@ export class ScopedResourceServer<T = object, K = string> implements ResourceSer
     return getBackendSrv().delete<MetaStatus>(`${this.url}/${name}`);
   }
 
-  private parseLabelSelector<T>(labelSelector: ListOptionsLabelSelector<T> | undefined): string | undefined {
+  // Allow overwriting the URL
+  // There are situations where we might want to enforce another URL: allowing users to provide their own API implementation
+  public overwriteUrl(url: string) {
```

Comment:
could we approach this more consistently?  rather than have each thing come up with its own approach to settings (the two URLs for scopes are 🤮 ), maybe something bootData that will retarget the URL based on group+version?+resource?  maybe something like:
```
interface APIServerURLOverride {
  url: string;
  group?: string;
  version?: string;
  group?: string;
}
```

and boot data could have:
```
apiServerURLOverrides?: APIServerURLOverride[]
```

Reply:
I wonder if there will be a lot of cases where users would actually need to bring their own external APIs instead of the ones provided within Grafana. We did it for scopes as we have an imminent requirement for it and the config file was the easiest approach but I really don't know how this will evolve over time. cc @bergquist.

On the other hand, I love this approach from a simplicity point of view although it might be confusing for some if they don't pay attention to the implementation of the ScopedResourceServer.

As a question from my side: how would one configure an external API? What would be the mechanism for that?

Reply:
I think this is fine @bfmatei. Not sure how wildly the overrides will be used in the future (maybe @ryantxu has some insights), but to provide a more scalable and user friendly way of defining overrides at scale I would expect them not to be part of the ini file, but probably a provisioning file (yaml). But we can introduce such extension point in the ini file in the future, as it will not change the front-end interface of the `APIServerURLOverride`  you are proposing in this PR.

Reply:
Alrighty, given the latest discussions we agreed on the following: remove the URL overwrite from this PR, write a design document about it and address the change in a subsequent PR.

The scopes feature does not require a URL overwrite just now.